### PR TITLE
fix(vite)

### LINKED ISSUE #4090: ❌ build issues: vitejs.dev=5.0.0
Running log of build failures for vitejs.dev=5.0.0

### LINKED ISSUE #4151: ❌ build issues: vitejs.dev=5.0.1
Running log of build failures for vitejs.dev=5.0.1

### LINKED ISSUE #4156: ❌ build issues: vitejs.dev=5.0.2
Running log of build failures for vitejs.dev=5.0.2

### PR DESCRIPTION
**Pull request by isworking**
closes #4090

### CODE REVIEW

**Comment by isworking** [comment]
@jhheider Any idea why this is happening? The binary `vite` is in `.pkgx/vitejs.dev/v*/bin`, but it's not working. Is it not in path or something?

**Comment by isworking** [comment]
The binary is right there (symlinked), but pkgx is unable to see it? It's weird. Also, it works locally, which is even weirder (both `pkgx vite --version` and `pkg test` work).

File: projects/vitejs.dev/package.yml [hunk 1]
```diff
@@ -1,22 +1,27 @@
-distributable: ~
+distributable: https://github.com/vitejs/vite/archive/refs/tags/v{{version.raw}}.tar.gz
 
 versions:
   github: vitejs/vite/tags
   strip: /v/
 
+dependencies:
+  pnpm.io: "*"
```
Comment (mxcl) [inline projects/vitejs.dev/package.yml]: this is not a runtime dependency surely?

File: projects/vitejs.dev/package.yml [hunk 2]
```diff
@@ -1,22 +1,27 @@
...
+  nodejs.org: ^18
```
Comment (mxcl) [inline projects/vitejs.dev/package.yml]: this is too strict, what's our strategy @jhheider ? `^16 || ^18 || ^20` or something right?

**Review by mxcl — CHANGES_REQUESTED** [review]

File: projects/vitejs.dev/package.yml (hunk 1 above)
Comment (isworking) [reply projects/vitejs.dev/package.yml]: Moved to `build`

File: projects/vitejs.dev/package.yml (hunk 2 above)
Comment (jhheider) [reply projects/vitejs.dev/package.yml]: yes... if the package.json defines engines we should mirror it.

**Comment by jhheider** [comment]
Great! Thanks for your persistence here, @Rajdeep-TG !

**Pull request merged**